docs/workloads/azure/backend/netcore/testing/provider_contract_testing.md

```diff
@@ -0,0 +1,146 @@
+---
+id: provider_contract_testing_dotnet
+title: Contract Testing - .Net Provider
+sidebar_label: Contract Testing
+hide_title: false
+hide_table_of_contents: false
+description: Dotnet provider contract tests
+keywords:
+  - dotnet api tests
+  - rest 
+  - api
+  - contract
+  - pact
+  - pactflow
+  - pipeline
+  - dotnet
+  - testing
+---
+
+
+## Contract Testing the API
+The [.Net WebAPI](../quickstart/web_api/create_project_netcore.md) Stack includes a Provider implementation of Bi-Directional Contract Testing to accelerate teams getting started with contract testing.
+
+### What is Bi-Directional Contract Testing
+Bi-Directional Contract Testing is a type of static contract testing where two contracts - one representing consumer expectations, and another representing the provider's capability - are compared to ensure they are compatible.
+
+> Bi-Directional contract testing replaces Consumer-Driven contract testing which is a more complicated but more thorough version of contract testing. Bi-Directional contract testing allows a provider to publish their OpenAPI specification instead of a Pact contract. When the provider and/or the consumer publish their contract, PactFlow will verify that the two contracts (consumer Pact and provider OpenAPI spec) are compatible.
+
+The general steps for a **Provider** in Bi-Directional contract testing are as follows:
+
+1. Create the provider contract (i.e. OpenAPI specification) either manually or generated by code
+2. (Optional) The provider contract is tested against the provider itself to ensure the API satisfies the contract
+3. The contract is published to PactFlow
+4. Run can-i-deploy to check the compatibility with its consumers
+
+In .Net WebAPI Stack, the OpenAPI specification is generated based upon the code, so there is no requirement for it to include verification tests (step 2) as we know the implementation and specification match.
+
+The following diagram shows an example flow of a Bi-Directional contract testing implementation with 1 provider and 1 consumer.
+
+![Bi-Directional Contract Testing](../../../../../../static/img/StacksQA_Bi-Directional-Contract-Testing.png)
+
+
+### Getting Started
+When creating a project using the [WebAPI template](../\quickstart\web_api\create_project_netcore.md), the `ci.yml` pipeline will include the steps required for Contract Testing as a provider.
+
+Within the contract testing steps, there are a number of variables that are required. In order for the pipeline to work as expected, the variable values will need to be updated before executing the pipeline.
+Variables can be found within `./build/azDevOps/azure/ci-vars.yml`
```

Comment:
I can't confirm the paths to say this bit as I can't see that file in the stacks-dotnet. has it yet to be merged in.  Also I notice that each offering has its own vars.yml so maybe need to update the file name.
Variables can be found within `./build/azDevOps/azure/ci-vars.yml`

![image](https://github.com/user-attachments/assets/200907a1-e95c-4fec-9db8-79410cc34a35)

Reply:
Yeah, I am working on a branch adding this.
I have created a project off of the .Net template and the vars are in that location

![image](https://github.com/user-attachments/assets/d98eb9f4-e038-4dbb-b8ff-12e0650e70b3)